### PR TITLE
lease: modify API and persist lease to disk

### PR DESCRIPTION
**Pull request by xiang90**
I still need to document the changed API and newly added functions.

But the code is almost done.

### CODE REVIEW

File: lease/lessor_test.go [hunk 1]
```diff
@@ -47,16 +63,21 @@ func TestLessorGrant(t *testing.T) {
 // the DeleteableKV.
 // The revoked lease cannot be got from Lessor again.
 func TestLessorRevoke(t *testing.T) {
+	dir, be := NewTestBackend(t)
+	defer be.Close()
+	defer os.RemoveAll(dir)
```
Comment (gyuho) [inline lease/lessor_test.go]: This shouldn't matter for testing, but would it be clearer to `Close` first and then `RemoveAll` like:

 ```
defer os.RemoveAll(dir)
defer be.Close()
```

?


Below, it removes first and close might fail silently?

 ```
defer be.Close()
defer os.RemoveAll(dir)
```

Comment (xiang90) [inline lease/lessor_test.go]: it does not really matter. OS will only release the resource when all the file descriptor are released. But yes, I can change to make this read better.

File: lease/lessor.go [hunk 2]
```diff
@@ -47,49 +52,66 @@ type lessor struct {
 	// We want to make Grant, Revoke, and FindExpired all O(logN) and
 	// Renew O(1).
 	// FindExpired and Renew should be the most frequent operations.
-	leaseMap map[uint64]*lease
+	leaseMap map[int64]*lease
```
Comment (heyitsanthony) [inline lease/lessor.go]: are these `int64`s LeaseIDs?

File: lease/lessor_test.go (hunk 1 above)
Comment (gyuho) [inline lease/lessor_test.go]: > OS will only release the resource when all the file descriptor are released. 

Good to know. Thanks!

File: lease/lessor.go (hunk 2 above)
Comment (xiang90) [inline lease/lessor.go]: yes. I plan to move the LeaseID typedef from storage pkg to lease pkg.

**Comment by xiang90** [comment]
@gyuho @heyitsanthony All fixed. My next step is to finish the recovery logic and then hook up this implementation with lease API.

**Comment by gyuho** [comment]
LGTM

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**